src/test/java/com/puppycrawl/tools/checkstyle/internal/ArchUnitCyclesCheckTest.java

```diff
@@ -0,0 +1,60 @@
+///////////////////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code and other text files for adherence to a set of rules.
+// Copyright (C) 2001-2022 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.internal;
+
+import static com.tngtech.archunit.library.dependencies.SlicesRuleDefinition.slices;
+
+import org.junit.jupiter.api.Test;
+
+import com.tngtech.archunit.core.domain.JavaClasses;
+import com.tngtech.archunit.core.importer.ClassFileImporter;
+import com.tngtech.archunit.core.importer.ImportOption;
+import com.tngtech.archunit.lang.ArchRule;
+import com.tngtech.archunit.library.freeze.FreezingArchRule;
+
+public class ArchUnitCyclesCheckTest {
+
+    /**
+     * Tests that there are no
+     * <a href="https://www.archunit.org/userguide/html/000_Index.html#_cycle_checks">cycles</a>
```

Comment:
`Tests that there are no cycles in the package` => `Tests that have no cycles in the package`

Reply:
Why? Isn't `Tests that there are no cycles in the package dependencies` correct?

Reply:
"that there" sounds very odd and I would say it is not proper. "That there" is more like you are pointing to a specific thing (noun) like "that there doggy in the window" but you don't really follow up with a noun to point to. The replacement text sounds more natural to me. "Tests which have" would also be acceptable.

https://brians.wsu.edu/2016/05/25/this-here-that-there/

Reply:
`Tests that have no`, in this `Test` is acting as a noun, similar to `Boys that have no`, whereas we should be using `Test` as a verb.

Reply:
Your original sentence is not using Test as a verb. It is also a noun.

Please suggest a sentence if you don't want to use my wording, but "that there" still has to change.

Reply:
Nick suggested `Detects cycles in package dependencies`, is that fine?

Reply:
Done.